File: src/kakao/bank2021/Solution5.java
package kakao.bank2021;

public class Solution5 {

    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];

        for (int i = 0; i < prices.length; i++) {
            // 해당 원소 뒤의 원소가 나보다 큰지 비교
            int count = 0;

            for (int j = i + 1; j < prices.length; j++) {
                if (prices[i] < prices[j]) count++;
            }

            answer[i] = count;
        }
        return answer;
    }
}

// 시간 부족할 거 같아서 최적화는 못함..
// 42 / 70

File: src/toss/Solution5.kt
package toss

/**
 * 부가가치세 계산기
 *
 * 과세 금액 10% 첫째자리 올림처리
 * 과세금액 = 공급가액 - 비과세금액
 * 공급가액 + 부가가치세 = 공급대가
 * 과세 금액 = 공급가액 - 비과세 금액
 * 대가 = 부가 + 비과세 금액 + 과세 금액
 * 주문금액
 */
class Solution5 {
    fun solution(orderAmount: Long, taxFreeAmount: Long, serviceFee: Long): Long {
        var 공급대가 = orderAmount
        if (serviceFee > 0) {
            공급대가 -= serviceFee
        }
        // 공급가액 + 부가가치세 = 공급대가
        // 과세 금액 = 공급가액 - 비과세 금액
        // 대가 = 부가 + 비과세 금액 + 과세 금액
        val 부가_플러스_과세금액 = 공급대가 - taxFreeAmount

        // orderAmount : 주문금액
        // taxFreeAmount : 비과세금액
        // serviceFee : 봉사료
        return 0
    }

    fun testTemplate(orderAmount: Long, taxFreeAmount: Long, serviceFee: Long, answer: Long) {
        val ret = solution(orderAmount, taxFreeAmount, serviceFee)
        if (answer == ret) {
            println("정답 입니다!!")
        } else {
            println("오답 입니다!!")
        }
    }
}

fun main() {
    Solution5().let {

    }
}